cam6_4_143: Extended non-LTE CO2 cooling parameterization

fvitt opened the pull request:
Closes #1358

fvitt committed:
add sparse checkout submodule ext_co2_cooling

fvitt committed:
add option to use extended co2 cooling method

	new file:   src/physics/waccm/nlte_extco2.F90
	modified:   bld/configure
	modified:   bld/namelist_files/namelist_definition.xml
	modified:   src/physics/waccm/nlte_lw.F90
	modified:   src/physics/waccm/radheat.F90

fvitt committed:
implement configure switch to avoid nag compiler builds

	modified:   bld/config_files/definition.xml
	modified:   bld/configure
	modified:   cime_config/testdefs/testmods_dirs/cam/outfrq9s_ctem/shell_commands
	modified:   cime_config/testdefs/testmods_dirs/cam/outfrq9s_ctem/user_nl_cam
	modified:   src/physics/waccm/nlte_extco2.F90
	modified:   src/physics/waccm/radheat.F90

fvitt committed:
update default WACCMX ne16pg3 IC file

cacraigucar commented on src/physics/waccm/nlte_extco2.F90:
```diff
@@ -0,0 +1,141 @@
+    real(r8), intent(in) :: co2_mw    ! CO2 molecular weight
+    real(r8), intent(in) :: n2_mw     ! N2 molecular weight
+
+#ifdef EXT_CO2_COOL
```
I'm wondering about why you felt the need to encapsulate this code within a #ifdef.  We try to avoid using these directives and backtracing this particular routine, it is only called when `nlte_use_extco2` is true.  I believe this #ifdef is not needed.   I've not analyzed the other #ifdef directives, but I would suggest there might also might be others that are not needed.  If all can be removed, please remove the definition from configure as well.

fvitt replied:
The external code ext_co2_cooling code does not compile with NAG compiler.

cacraigucar replied:
With that in mind, does it make sense to only #ifdef the calls and the use statements (keep it contained around the code which only needed to not compiled)?

fvitt replied:
I would prefer not to add the history fields when the model is not configured to use the extended cooling method.

cacraigucar replied:
Is there something preventing you from passing in the variable `nlte_use_extco2` and using it to determine if the history fields are being set or not?

fvitt replied:
I have concluded with the checks in higher level modules, the cpp ifdefs in init routine was not necessary.

cacraigucar commented on bld/namelist_files/namelist_defaults_cam.xml:
```diff
@@ -182,7 +182,7 @@
 <ncdata hgrid="ne16np4"  nlev="126" waccmx="1" aquaplanet="1"                  >atm/waccm/ic/waccmx_aqua_ne16np4_126L_c191108.nc</ncdata>
 <ncdata hgrid="ne16np4"  nlev="126" waccmx="1" aquaplanet="1" ionosphere="none">atm/waccm/ic/waccmx4_neutral_aquap_ne16np4_126lev_c200827.nc</ncdata>
-<ncdata hgrid="ne16np4"  nlev="130" waccmx="1"     >atm/waccm/ic/waccmx_ne16pg3_L130_c250318.nc</ncdata>
+<ncdata hgrid="ne16np4"  nlev="130" waccmx="1"     >atm/waccm/ic/FX2000_ne16pg3_spinup01.cam.i.0002-01-01_c251215.nc</ncdata>
```
Has this file been "rimport"ed?

fvitt replied:
done

cacraigucar commented on bld/namelist_files/namelist_definition.xml:
```diff
@@ -7994,6 +7994,13 @@ Call ALI-ARMS every X timesteps
 
+<entry id="nlte_use_extco2" type="logical" category="waccm_phys"
+       group="radheat_nl" valid_values="" >
+If TRUE, then use the extended CO2 scheme of Lopez-Puertas et al. 2024 as an alernative method of computing non-LTE
```
Since this is a comment which will show up on a web page, would it be appropriate to include the DOI here also?

fvitt replied:
DOI has been added here

fvitt committed:
add DOI reference, set co2cooling to zero outside cpp ifdef

	modified:   bld/namelist_files/namelist_definition.xml
	modified:   src/physics/waccm/nlte_extco2.F90

fvitt committed:
adjust cpp ifdefs

	modified:   src/physics/waccm/nlte_extco2.F90

fvitt committed:
Merge remote-tracking branch 'ESCOMP/cam_development' into nonLTE_CO2_cool

fvitt committed:
ChangeLog draft

fvitt committed:
ChangeLog updates

fvitt committed:
adjust reg tests

fvitt committed:
ChangeLog update
